collect_app/src/main/java/org/odk/collect/android/widgets/RatingWidget.java

```diff
@@ -17,69 +17,89 @@
+    public RatingWidget(Context context, QuestionDetails questionDetails) {
         super(context, questionDetails);
-        binding.ratingBar.setNumStars(rangeQuestion.getRangeEnd().intValue());
-        binding.ratingBar.setStepSize(1.0F);
     }
 
     @Override
     protected View onCreateAnswerView(Context context, FormEntryPrompt prompt, int answerFontSize) {
-        binding = RatingWidgetAnswerBinding.inflate(((Activity) context).getLayoutInflater());
+        RatingWidgetAnswerBinding binding = RatingWidgetAnswerBinding.inflate(((Activity) context).getLayoutInflater());
         View answerView = binding.getRoot();
 
-        binding.ratingBar.setOnRatingBarChangeListener((ratingBar, rating, fromUser) -> widgetValueChanged());
-        binding.ratingBar.setEnabled(!prompt.isReadOnly());
+
+        RangeQuestion rangeQuestion = (RangeQuestion) prompt.getQuestion();
+        int numberOfStars = rangeQuestion.getRangeEnd().intValue();
+        int columns = calculateColumns(48);
+
+        int rows = (int) Math.ceil((double) numberOfStars / columns);
+
+        if (rows == 1) {
+            binding.ratingBar.setVisibility(VISIBLE);
+            this.ratingBar = binding.ratingBar;
+        } else {
+            binding.ratingBar.setVisibility(GONE);
+            columns = calculateColumns(36);
+            rows = (int) Math.ceil((double) numberOfStars / columns);
+            if (rows == 1) {
+                binding.ratingBar2.setVisibility(VISIBLE);
+                this.ratingBar = binding.ratingBar2;
+            } else {
+                binding.ratingBar3.setVisibility(VISIBLE);
+                this.ratingBar = binding.ratingBar3;
+            }
+        }
+
+        ratingBar.setNumStars(rangeQuestion.getRangeEnd().intValue());
+        ratingBar.setOnRatingBarChangeListener((ratingBar, rating, fromUser) -> widgetValueChanged());
+        ratingBar.setEnabled(!prompt.isReadOnly());
 
         if (prompt.getAnswerText() != null) {
-            binding.ratingBar.setRating(Integer.parseInt(prompt.getAnswerText()));
+            ratingBar.setRating(Integer.parseInt(prompt.getAnswerText()));
         }
         return answerView;
     }
 
     @Override
     public void setOnLongClickListener(OnLongClickListener l) {
-        binding.ratingBar.setOnLongClickListener(l);
+        ratingBar.setOnLongClickListener(l);
     }
 
     @Override
     public IAnswerData getAnswer() {
-        return binding.ratingBar.getRating() == 0.0F ? null : new IntegerData((int) binding.ratingBar.getRating());
+        return ratingBar.getRating() == 0.0F ? null : new IntegerData((int) ratingBar.getRating());
     }
 
     @Override
     public void clearAnswer() {
-        binding.ratingBar.setRating(0.0F);
+        ratingBar.setRating(0.0F);
     }
 
-    protected RatingWidgetAnswerBinding getBinding() {
-        return binding;
+    protected RatingBar getRatingBar() {
```

Comment:
This is just for tests right? I would change access modifier (of the field) to package-private and use that field directly without any getter method like this one.

Reply:
Thanks @grzesiek2010! That's a better approach.